File: test_terminal.js
#!/usr/bin/env node

// Test script to verify terminal functionality
import { execSync } from 'child_process';

console.log('🧪 Testing Terminal Environment...\n');

const tests = [
  {
    name: 'Node.js availability',
    command: 'node --version',
    expected: 'should return Node.js version'
  },
  {
    name: 'npm availability',
    command: 'npm --version',
    expected: 'should return npm version'
  },
  {
    name: 'npx availability',
    command: 'npx --version',
    expected: 'should return npx version'
  },
  {
    name: 'Basic shell commands',
    command: 'echo "Hello World"',
    expected: 'should output "Hello World"'
  },
  {
    name: 'Directory listing',
    command: 'ls -la',
    expected: 'should list directory contents'
  },
  {
    name: 'Current directory',
    command: 'pwd',
    expected: 'should show current working directory'
  },
  {
    name: 'Directory navigation',
    command: 'cd .. && pwd && cd -',
    expected: 'should navigate directories and return to original'
  },
  {
    name: 'File operations',
    command: 'touch test_file.txt && ls test_file.txt && rm test_file.txt',
    expected: 'should create, list, and remove test file'
  },
  {
    name: 'Python availability',
    command: 'python3 --version || python --version',
    expected: 'should return Python version'
  },
  {
    name: 'Git availability',
    command: 'git --version',
    expected: 'should return Git version'
  },
  {
    name: 'Environment variables',
    command: 'echo $PATH',
    expected: 'should show PATH environment variable'
  },
  {
    name: 'Shell features',
    command: 'echo "Test" | grep "Test"',
    expected: 'should demonstrate pipe functionality'
  }
];

let passed = 0;
let failed = 0;

for (const test of tests) {
  try {
    console.log(`🔍 Testing: ${test.name}`);
    console.log(`   Command: ${test.command}`);
    
    const result = execSync(test.command, { 
      encoding: 'utf8',
      cwd: process.cwd(),
      timeout: 5000
    }).trim();
    
    console.log(`   ✅ Result: ${result}`);
    console.log(`   📝 ${test.expected}\n`);
    passed++;
    
  } catch (error) {
    console.log(`   ❌ Failed: ${error.message}`);
    console.log(`   📝 ${test.expected}\n`);
    failed++;
  }
}

console.log('📊 Test Results:');
console.log(`   ✅ Passed: ${passed}`);
console.log(`   ❌ Failed: ${failed}`);
console.log(`   📈 Success Rate: ${Math.round((passed / (passed + failed)) * 100)}%`);

if (failed === 0) {
  console.log('\n🎉 All tests passed! Terminal environment is ready.');
  process.exit(0);
} else {
  console.log('\n⚠️  Some tests failed. Check your environment setup.');
  process.exit(1);
}

File: test_terminal.py
#!/usr/bin/env python3
"""
Test script for terminal executor
Run this to verify the backend terminal is working
"""

import sys
import os
sys.path.append(os.path.dirname(__file__))

from terminal_executor import terminal_executor

def test_basic_commands():
    """Test basic terminal commands"""
    print("Testing terminal executor...")
    
    # Test 1: Simple echo command
    print("\n1. Testing echo command:")
    result = terminal_executor.execute_command({
        "command": "shell",
        "session_id": "test-session-1",
        "command_line": "echo 'Hello World'"
    })
    print(f"Result: {result}")
    
    # Test 2: npm version
    print("\n2. Testing npm -v:")
    result = terminal_executor.execute_command({
        "command": "shell",
        "session_id": "test-session-2",
        "command_line": "npm -v"
    })
    print(f"Result: {result}")
    
    # Test 3: node version
    print("\n3. Testing node -v:")
    result = terminal_executor.execute_command({
        "command": "shell",
        "session_id": "test-session-3",
        "command_line": "node -v"
    })
    print(f"Result: {result}")
    
    # Test 4: python version
    print("\n4. Testing python --version:")
    result = terminal_executor.execute_command({
        "command": "shell",
        "session_id": "test-session-4",
        "command_line": "python --version"
    })
    print(f"Result: {result}")
    
    # Test 5: List files
    print("\n5. Testing list files:")
    result = terminal_executor.execute_command({
        "command": "list_files",
        "cwd": ""
    })
    print(f"Result: {result}")

if __name__ == "__main__":
    test_basic_commands()
